kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/config/Features.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.config;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import io.kroxylicious.proxy.config.Configuration;
+
+import static java.util.Arrays.stream;
+
+/**
+ * Represents the entire set of proxy features.
+ */
+public class Features {
+
+    private static final Features DEFAULT_FEATURES = new Features(Map.of());
```

Comment:
```suggestion
    private static final Features ENABLED_BY_DEFAULT = new Features(Map.of());
```
Though maybe thats a bit funky as it implies the proxy does nothing by default...

Reply:
Each Feature knows if it is enabled by default or not, so you can have a Feature that is default off and has to be enabled (like test-only configuration), but you could also have a feature that is enabled by default.

The default set of features uses the default enabled-ness of each feature.